Public/Src/Utilities/Collections/ConcurrentBigMapEntry.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace BuildXL.Utilities.Collections
+{
+    /// <summary>
+    /// This is a key value pair for the concurrent big map.
+    /// </summary>
+    /// <remarks>
+    /// We can't use the built-in KeyValuePair because it does not implement GetHashCode and Equals
+    /// </remarks>
+    public struct ConcurrentBigMapEntry<TKey, TValue> : IEquatable<ConcurrentBigMapEntry<TKey, TValue>>

```

Comment:
> This change uses a proper struct that can safely and performently be used as entries in hash based collections.

How is this struct more proper or more safe or more performant than `KeyValuePair`?  Is it all because this one implements Equals/HashCode?

Reply:
Correct, and it is not about safety, it is about perf due to nr of allocations.

Reply:
updated comment to reflect

Reply:
Maybe add that comment to the remarks as well.